src/components/modals/ReviewStakeModal.tsx

```diff
@@ -0,0 +1,176 @@
+import { AoGatewayWithAddress, IOToken } from '@ar.io/sdk/web';
+import { log, WRITE_OPTIONS } from '@src/constants';
+import { useGlobalState } from '@src/store';
+import { formatAddress, formatWithCommas } from '@src/utils';
+import { ArweaveTransactionID } from '@src/utils/ArweaveTransactionId';
+import { showErrorToast } from '@src/utils/toast';
+import { useQueryClient } from '@tanstack/react-query';
+import { useState } from 'react';
+import Button, { ButtonType } from '../Button';
+import { LinkArrowIcon } from '../icons';
+import LabelValueRow from '../LabelValueRow';
+import BaseModal from './BaseModal';
+import BlockingMessageModal from './BlockingMessageModal';
+import SuccessModal from './SuccessModal';
+import WithdrawWarning from './WithdrawWarning';
+
+const ReviewStakeModal = ({
+  gateway,
+  amountToStake,
+  onSuccess,
+  onClose,
+  walletAddress,
+  ticker,
+}: {
+  gateway: AoGatewayWithAddress;
+  amountToStake: number;
+  walletAddress: ArweaveTransactionID;
+  onClose: () => void;
+  onSuccess: () => void;
+  ticker: string;
+}) => {
+  const queryClient = useQueryClient();
```

Comment:
Question: whats the difference for us acquiring the current queryClient from the hook vs the constant?

Reply:
It's so the components can be reusable across projects and don't depend on a concrete instance.